commit 9f52a3bc12fb2bbbbe614c11a2a9b3152e8152c6 (refs/changes/86/1686/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-03-17T13:29:20-07:00 (2 years, 7 months ago)
    
    CNS-198 Peer version failure neglects to call callback; leaks redis handle

diff --git a/lib/dns-server.js b/lib/dns-server.js
index 03510b1..5cc3451 100644
--- a/lib/dns-server.js
+++ b/lib/dns-server.js
@@ -1083,6 +1083,7 @@ DNSServer.prototype.handleTransfer = function (q, z, cb) {
 						log.debug(perr,
 						    'unable to detect peer ' +
 						    'version');
+						ccb(perr);
 						return;
 					}
 					_.redis.set('peer:' +
